packages/jest-snapshot/src/utils.js

```diff
@@ -185,6 +185,15 @@ export const deepMerge = (target: any, source: any) => {
       if (isObject(source[key]) && !source[key].$$typeof) {
         if (!(key in target)) Object.assign(mergedOutput, {[key]: source[key]});
         else mergedOutput[key] = deepMerge(target[key], source[key]);
+      } else if (Array.isArray(source[key])) {
+        // Convert arrays to objects, merge, convert back to array
```

Comment:
Sounds expensive. Can we instead make `deepMergeArray` similar to what we have in `deepCyclicCopy.js`?
https://github.com/facebook/jest/blob/fb61bfffb803bad0c2ff9ab1798a6c4c31cb14fa/packages/jest-util/src/deepCyclicCopy.js#L83-L105